File: KotlinLab/ArrayOperations.java
import java.util.*;
import java.util.function.BiFunction;
import java.util.stream.Collectors;

class ArrayOperations {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Введите количество элементов массива: ");
        int n = scanner.nextInt();

        int[] array = new int[n];
        System.out.println("Введите элементы массива: ");
        for (int i = 0; i < n; i++) {
            array[i] = scanner.nextInt();
        }

        int sum = arrayOp(array, 0, Integer::sum);
        int product = arrayOp(array, 1, (a, b) -> a * b);
        int min = arrayOp(array, Integer.MAX_VALUE, Math::min);
        int max = arrayOp(array, Integer.MIN_VALUE, Math::max);

        System.out.println("Сумма: " + sum);
        System.out.println("Произвведение: " + product);
        System.out.println("Минимальный элемент массива: " + min);
        System.out.println("Максимальный элемент массива: " + max);

        int mostFrequent = mostFreqElem(array);
        System.out.println("Наиболее часто встречающийся элемент: " + mostFrequent);

        List<Integer> elemEvenOccElem = elemEvenOcc(array);
        System.out.println("Четные элементы, повторяющиеся четное количество раз: " + elemEvenOccElem);

        List<Integer> posNumSumGreatTenList = posNumSumGreatTen(array);
        System.out.println("Отфильтрованный список (удалены отрицательные числа с суммой цифр < 10): " + posNumSumGreatTenList);
    }

    public static int arrayOp(int[] array, int initialValue, BiFunction<Integer, Integer, Integer> func) {
        int result = initialValue;
        for (int value : array) {
            result = func.apply(result, value);
        }
        return result;
    }

    public static int mostFreqElem(int[] array) {
        Map<Integer, Long> frequencyMap = Arrays.stream(array).boxed()
            .collect(Collectors.groupingBy(e -> e, Collectors.counting()));

        return Collections.max(frequencyMap.entrySet(), Map.Entry.comparingByValue()).getKey();
    }

    public static List<Integer> elemEvenOcc(int[] array) {
        Map<Integer, Long> frequencyMap = Arrays.stream(array).boxed()
            .filter(e -> e % 2 == 0)
            .collect(Collectors.groupingBy(e -> e, Collectors.counting()));

        return frequencyMap.entrySet().stream()
            .filter(entry -> entry.getValue() % 2 == 0)
            .map(Map.Entry::getKey)
            .collect(Collectors.toList());
    }

    public static List<Integer> posNumSumGreatTen(int[] array) {
			return Arrays.stream(array).boxed()
					.filter(e -> !(e < 0 || digitSum(e) > 10))
					.collect(Collectors.toList());
		}

    private static int digitSum(int number) {
        int sum = 0;
        while (number != 0) {
            sum += number % 10;
            number /= 10;
        }
        return sum;
    }
}

File: KotlinLab/ArrayOperations.kt
import java.util.*
import kotlin.collections.HashSet

fun main() {
    val scanner = Scanner(System.`in`)

    print("Введите количество элементов массива: ")
    val n = scanner.nextInt()

    val array = IntArray(n)
    println("Введите элементы массива: ")
    for (i in 0 until n) {
        array[i] = scanner.nextInt()
    }

    val sum = arrayOp(array, 0) { a, b -> a + b }
    val product = arrayOp(array, 1) { a, b -> a * b }
    val min = arrayOp(array, Int.MAX_VALUE) { a, b -> minOf(a, b) }
    val max = arrayOp(array, Int.MIN_VALUE) { a, b -> maxOf(a, b) }

    println("Сумма: $sum")
    println("Произведение: $product")
    println("Минимальный элемент массива: $min")
    println("Максимальный элемент массива: $max")

    val mostFreqElem = mostFreqElem(array.toList())
    println("Самый частый элемент: $mostFreqElem")

    val elemEvenOcc = elemEvenOcc(array.toList())
    println("Четные элементы, повторяющиеся четное количество раз: $elemEvenOcc")

    val posNumSumGreatTen = posNumSumGreatTen(array.toList())
    println("Список без отрицательных чисел, сумма цифр которых меньше 10: $posNumSumGreatTen")
    println("-------------------------------------------------------------------------------")
    val m: MutableList<User> = mutableListOf()
    val u1=User("Фам","Им","От","@a", Date(1999,7,2))
    val u2=User("Иванов","Иван","Иванович","@ivan", Date(1964,2,12))
    val u3=User("Василий","Пупкин","","@lesnik", Date(1988,10,30))
    val u4=User("Иванов","Иван","Иванович","@ivan", Date(1964,2,12))
    m.add(u1)
    m.add(u2)
    m.add(u3)
    m.add(u4)
    for (x in m)x.write()
    println("-------------------------------------------------------------------------------")
    m.sort()
    for (x in m)x.write()
    println("-------------------------------------------------------------------------------")
    val h: HashSet<User> = hashSetOf()
    h.add(u1)
    h.add(u2)
    h.add(u3)
    h.add(u4)
    for (x in h)x.write()
}

fun arrayOp(array: IntArray, initialValue: Int, func: (Int, Int) -> Int): Int {
    var result = initialValue
    for (value in array) {
        result = func(result, value)
    }
    return result
}

fun mostFreqElem(list: List<Int>): Int? {
    return list.groupingBy { it }.eachCount().maxByOrNull { it.value }?.key
}

fun elemEvenOcc(list: List<Int>): List<Int> {
    return list.groupingBy { it }
        .eachCount()
        .filter { (key, value) -> key % 2 == 0 && value % 2 == 0 }
        .keys.toList()
}

fun posNumSumGreatTen(list: List<Int>): List<Int> {
    return list.filter { number ->
        !(number < 0 || number.toString().sumOf { Character.getNumericValue(it) } > 10)
    }
}